sound/soc/codecs/rt715.c

```diff
@@ -498,6 +502,7 @@ static int rt715_set_bias_level(struct snd_soc_component *component,
 			regmap_write(rt715->regmap,
 						RT715_SET_AUDIO_POWER_STATE,
 						AC_PWRST_D0);
+			msleep(power_up_delay);
```

Comment:
a kernel parameter is nice but should there be a control as well?

Reply:
@plbossart  why do we need a control for this parameter? We may only need to set this parameter once at alsa-base,conf depends on different project or dmic.  Please let us know your concern, thanks!

Reply:
@jack-cy-yu a kernel parameter will be used for all usages of rt715. The control enables different values to be chosen for different platforms. It's just more flexible.

Reply:
@plbossart  do you mean to set up a kcontrol like below pseudo code to select different power_up_delay? 

static int rt715_pwr_delay_put(struct snd_kcontrol *kcontrol,
		struct snd_ctl_elem_value *ucontrol)
{
	...
	if (rt715->power_delay == 1)
		power_up_delay = 400;
	else if (rt715->power_delay == 2)
		power_up_delay = 500;
	....
}

Reply:
Something like that yes, with maybe clearer values (the mapping 1->400, 2->500) is not very obvious.

But if you think in most cases a single value is good enough, then the kernel parameter is fine.

Reply:
Yes, I think a single value is good enough for most cases.

Reply:
ok then, let's keep it this way